File: src/main/java/com/example/digest/models/Newsletter.java
package com.example.digest.models;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;

import java.time.LocalDateTime;

@Entity
@Table(name = "newsletters")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Newsletter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "article_count", nullable = false)
    private Integer articleCount;

    @Column(name = "status", nullable = false)
    private String status; // GENERATED, SENT, FAILED

    @CreatedDate
    @Column(name = "generated_at", nullable = false)
    private LocalDateTime generatedAt;

    @Column(name = "sent_at")
    private LocalDateTime sentAt;
}

File: handlers/shared/models/newsletter.ts
export interface Newsletter {
  id: string
  title: string
  articleCount: number
  status: NewsletterStatus
  generatedAt: string
  sentAt?: string
  htmlS3Key: string
  errorMessage?: string
}

export enum NewsletterStatus {
  GENERATED = 'GENERATED',
  SENT = 'SENT',
  FAILED = 'FAILED',
}
